melodic/doc-build.yaml

```diff
@@ -0,0 +1,58 @@
+%YAML 1.1
+# ROS buildfarm doc-build file
+---
+canonical_base_url: http://docs.ros.org
+documentation_type: rosdoc_lite
+jenkins_job_priority: 89
```

Comment:
The priorities shouldn't be equal but give preference to newer distros. It might be necessary to shuffle priorities to make room for the new ones.

Same below for the other build files.

Reply:
I've shuffled around the priorities in 192e632 and bc96a4a.  I think I've got everything in the same relative priority order, with melodic inserted as appropriate.  A look at the output of `list_priorities.py` from someone else would be helpful, though.

Reply:
The Jenkins master is currently provisioned to only support priorities from 1 to 99.

Reply:
> The Jenkins master is currently provisioned to only support priorities from 1 to 99.

Ah, that stinks :(.  OK, will collapse the numbers down.

Reply:
Maybe we need to make more room. Or change some of the priorities of the maintenance jobs. We should just make sure to have enough room for the next several releases to not run into the problem next year again.

Reply:
There is actually quite a bit of the range we are not using (our lowest is 58, but even if we are leaving room for other things, we can go way lower than that).  I'll move the low end much further down, and insert some space between groupings so we have some breathing room.

Reply:
The maintenance jobs are using priorities in the lower range. And they need to have a higher prio then the package specific jobs. So you can't just use smaller numbers. See https://github.com/ros-infrastructure/ros_buildfarm/search?utf8=%E2%9C%93&q=priority%3D&type=

Reply:
What is the policy for EOL distros? Do we need to keep the EOL jobs priorities in orde ? Or can we just reassign them? this way if we never have more than 4 distros we don't have to modify the range

Reply:
That would require to shift the priorities of all existing distros which is a significant reconfiguration effort I would rather avoid. We should just make sure that we have a large enough range for future releases (e.g. have "room" for the next ten year - by then the infrastructure will likely have been changed anyway).

Reply:
That's true.  At the very least, I can collapse the "jade" priorities into the "indigo" numbers, for instance; since we never have jade jobs, that shouldn't really be a problem.

(I think we should probably just remove jade completely, but that is a different PR)

Reply:
> We should just make sure that we have a large enough range for future releases (e.g. have "room" for the next ten year - by then the infrastructure will likely have been changed anyway).

The problem is that we have 6 "classes" of jobs: PR priority, commit priority, source priority, basic binary priority, doc priority, and extended binary priority.  If we want to leave room for 6 more years (say), we would need at least 60 spots.  But because of the pressure on the lower end from maintenance jobs, I think we only have 50 (the highest maintenance priority I see in a simple search is 40; I'm leaving the whole 40 range empty just for safety).

The nicest thing would be to allow the buildfarm to have more priority numbers.  How hard is it to change it?

Reply:
We can change that easily in the settings but that should be reflected in the provisioning too. Changing priorities of existing jobs once is ok (we might not get around it). But that change should be future proof.

Reply:
> We can change that easily in the settings but that should be reflected in the provisioning too. Changing priorities of existing jobs once is ok (we might not get around it). But that change should be future proof.

Makes sense.  I think we have to change them now to fit into ranges, so I'll go ahead and do that with the understanding that we'll need to change the buildfarm before we deploy this.